docs/specification/draft/basic/authentication.md

```diff
@@ -0,0 +1,142 @@
+---
+title: Authentication
+type: docs
+weight: 15
+---
+
+{{< callout type="info" >}}
+**Protocol Revision**: {{< param protocolRevision >}}
+{{< /callout >}}
+
+## Overview
+
+The Model Context Protocol currently supports authentication through OAuth 2.0. Authentication is only available when using an HTTP transport (such as SSE). Authentication occurs at the transport level. This specification defines how clients authenticate with MCP servers using standard OAuth 2.0 flows.
+
+## Authentication Flow
+
+MCP servers **MAY** require authentication for any request. When authentication is required and has not been provided, servers **MUST** respond with *HTTP 401 Unauthorized*. Clients **MUST** then initiate the OAuth 2.0 authentication flow.
+
+The basic authentication flow is as follows:
+
+```mermaid
+sequenceDiagram
+    participant C as Client
+    participant M as MCP Server
+    participant A as OAuth Server
+
+    C->>M: Any MCP Request
+    M->>C: HTTP 401 Unauthorized
+    C->>M: GET /authorize?redirect_uri=http://localhost:1234/callback
+    Note over M: Validate redirect_uri
+    M->>C: Redirect to OAuth Server
+    C->>A: OAuth Authorization Request
+    A->>M: Authorization Code (via redirect)
+    M->>A: Exchange Code for Tokens
+    A->>M: Access Token + Refresh Token
+    M->>C: Redirect to validated redirect_uri with session token
+    C->>M: Subsequent requests with session token
```

Comment:
Possibly something to consider here, depending on whether keeping the SSE transport browser-compatible is a goal: `EventSource` doesn't support setting additional headers, at least in the browser. (It _does_ support sending `Cookie` headers automatically using the `{ withCredentials }` `EventSourceInit` option, however.)

Reply:
This is a really good point. 😕 

I don't love cookies as a replacement here. Possibly the most correct answer is, "just use WebSocket," but of course that has its own problems…

Reply:
Maybe the use of a package like [eventsource](https://www.npmjs.com/package/eventsource) is an acceptable workaround.

Reply:
Used `eventsource` in this PR: https://github.com/modelcontextprotocol/typescript-sdk/pull/134